recipes/vllm/recipe.yaml

```
@@ -0,0 +1,192 @@
+
+source:
+- url: https://pypi.org/packages/source/v/vllm/vllm-${{ version }}.tar.gz
+  sha256: bdeeda5624182e6a93895cbb7e20b6e88b04d22b8272d8a255741b28b36ae941
+  patches:
+  - patches/vllm-cmakefiles.patch
+  - if: linux and use_cuda
+    then:
+    - patches/vllm-cpu-utils.patch
+  - if: is_cross_compiling
+    then:
+    - patches/vllm-cmake-args.patch
+  target_directory: vllm
+- url: https://github.com/vllm-project/flash-attention/archive/720c94869cf2e0ff5a706e9c7f1dce0939686ade.tar.gz
+  sha256: 5ba61742ebbf496d9daa846ed09b51f6e941db955398be75e2565141e2656219
+  target_directory: flash-attention
+
+build:
+  number: 0
+  script: |
+    sed -i.bak 's/set(TORCH_SUPPORTED_VERSION_CUDA "2.4.0")/set(TORCH_SUPPORTED_VERSION_CUDA "2.5.1")/g' flash-attention/CMakeLists.txt
+    export VLLM_FLASH_ATTN_SRC_DIR=$SRC_DIR/flash-attention
+    cd vllm
+    python use_existing_torch.py
+    mkdir -p $SRC_DIR/vllm/third_party/NVTX/c
+    ln -s $PREFIX/include $SRC_DIR/vllm/third_party/NVTX/c/include
+    # export VERBOSE=1
+    export VLLM_TARGET_DEVICE=${{ vllm_target_device }}
+    ${{ PYTHON }} -m pip install . --no-build-isolation
+
+  python:
+    entry_points:
+    - vllm  =  vllm.scripts:main
+
+  skip: win or (osx and x86_64)
+
+requirements:
+  build:
+  - ninja
+  - cmake
+  - git
+  - zlib
+  - ${{ stdlib('c') }}
+  - ${{ compiler('c') }}
+  - ${{ compiler('cxx') }}
+  - if: use_cuda
+    then:
+    - ${{ compiler('cuda') }}
+  - if: is_cross_compiling
+    then:
+    - python
+    - cross-python_${{ target_platform }}
+    - pytorch ==${{ pytorch_version }}
+    - if: use_cuda
+      then:
+      - pytorch-gpu
+      else:
+      - pytorch-cpu
+  host:
+  - python
+  - pip
+  - setuptools
+  - setuptools-scm
+  - packaging
+  - wheel
+  - jinja2
+  - pytorch ==${{ pytorch_version }}
+  - if: linux
+    then:
+    - libnuma
+  - if: use_cuda
+    then:
+    - pytorch-gpu
+    - nvtx-c
+    - cuda-version ==${{ cuda_compiler_version }}
+    - if: is_cuda_12
+      then:
+      - cuda
+      - cuda-cudart-dev
+      - cuda-nvrtc-dev
+      - cuda-nvrtc-static
+      - libcublas-dev
+    - cutlass
+    else:
+    - pytorch-cpu
+  run:
+  - python >=${{ python_min }}
+  - psutil
+  - sentencepiece
+  - numpy <2.0.0
+  - requests >=2.26.0
+  - tqdm
+  - blake3
+  - py-cpuinfo
+  - transformers >=4.48.2
+  - tokenizers >=0.19.1
+  - protobuf
+  - fastapi <0.113.0,>=0.107.0
+  - aiohttp
+  - openai >=1.52.0
+  - uvicorn-standard
+  - pydantic >=2.9
+  - prometheus_client >=0.18.0
+  - pillow
+  - prometheus-fastapi-instrumentator >=7.0.0
+  - tiktoken >=0.6.0
+  - lm-format-enforcer <0.11,>=0.10.9
+  - outlines ==0.1.11
+  - lark ==1.2.2
+  - if: x86_64
+    then:
+    - xgrammar >=0.1.6
+    - pybind11
+  - typing_extensions >=4.10
+  - filelock >=3.16.1
+  - partial-json-parser
+  - pyzmq
+  - msgspec
+  - gguf ==0.10.0
+  - importlib-metadata
+  - mistral-common >=1.5.0
+  - opencv-python-headless >=4.0.0  # mistral-common[opencv]
```

```suggestion
  - opencv >=4.0.0  # mistral-common[opencv]
```

if you don't want qt to installed at build time you can:

1. Depend on `noqt` https://github.com/conda-forge/noqt-feedstock
2. Depend on `opencv =*=headless*`

---

as a runtime dependency, just depend on opencv, 

people have historically come to opencv from conda-forge as a "batteries included" package.

We've provided options for system integrators to install headless ,but by default it should still install the "qt version" for linux + windows users.

---

Sounds good! Thanks for your feedback!